### ISSUE TITLE
tar files now generate warnings when extracting

### ISSUE THREAD

**Issue by jonahgraham** [eclipse-platform/eclipse.platform.releng#147]
When I extract eclipse-platform-4.26M3-linux-gtk-x86_64.tar.gz I get a bunch of warnings. I don't get these warnings with 4.25 nor 4.26 M2. As every file and folder generates this warning, with the SDK tar that means > 1000 warnings.

```sh
$ tar xf ~/Downloads/eclipse-platform-4.26M3-linux-gtk-x86_64.tar.gz 
tar: Ignoring unknown extended header keyword 'LIBARCHIVE.creationtime'
tar: Ignoring unknown extended header keyword 'LIBARCHIVE.creationtime'
tar: Ignoring unknown extended header keyword 'LIBARCHIVE.creationtime'
tar: Ignoring unknown extended header keyword 'LIBARCHIVE.creationtime'
...

$ tar --version
tar (GNU tar) 1.30
Copyright (C) 2017 Free Software Foundation, Inc.
```

I am on Ubuntu (Xubuntu specifically) 20.04.

**Comment by jonahgraham** [comment]
I don't know how important this warning is, nor how widespread it may be. A quick google indicates this is normal for tar files built on macOS and extracted on Ubuntu. But we don't build on macOS right?

**Comment by akurtakov** [comment]
All archives should be build on Linux. @SDawley , would you please investigate?

**Comment by SDawley** [comment]
Update: We build on centos8, [eclipsecbi/jiro-agent-centos-8-jdk11:latest](https://github.com/eclipse-cbi/jiro-agents) specifically.

That said, I also get these errors. They can be fixed by using the `--no-xattrs` option, which is available on GNU tar so I will try adding that to the build.

**Comment by fredg02** [comment]
Just a data point:

I see the same messages as Jonah with tar v1.34 on Kubuntu 22.04.1 LTS.

**Comment by jonahgraham** [comment]
I think this is a tycho or somewhere else upstream issue. I have moved this issue up the stack in https://github.com/eclipse-tycho/tycho/issues/2586

**Comment by laeubi** [comment]
> As every file and folder generates this warning, with the SDK tar that means > 1000 warnings.

Maybe one can report this to the `tar` project that such warnings are not printed repeatably...

**Comment by merks** [comment]
Somewhere I read that `--warning=no-unknown-keyword` can suppress this warning.  That will probably have to suffice as far as tar providing anything is concerned.  I also read a suggestion to use `--no-xattrs` in order to avoid to create these extended things in the first place.  I wonder if Tycho uses a tar library for which it could set such an attribute to avoid this at the source?

**Comment by laeubi** [comment]
> That will probably have to suffice as far as tar providing anything is concerned

I don't think so, printing the same warning more than once seems not very useful and is the root cause here, a user might want to see warnings about ignored extensions but obviously not a thousand times without any new information.

>  I also read a suggestion to use --no-xattrs in order to avoid to create these extended things in the first place.

This is a `tar` option that obviously not applicable for Tycho, also just because `tar` can't read the attribute there must be some other tools using it somewhere or otherwise no one would write such extensions. So another option would be to simply tell `tar` how to interpret (or silently ignore) such extension.

> I wonder if Tycho uses a tar library for which it could set such an attribute to avoid this at the source?

See https://github.com/eclipse-tycho/tycho/issues/2586#issuecomment-1615942418
